clusters/resource_cluster.go

```diff
@@ -241,14 +241,54 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *commo
 	common.DataToStructPointer(d, clusterSchema, &cluster)
 	var clusterInfo ClusterInfo
 	var err error
+
 	if hasClusterConfigChanged(d) {
 		log.Printf("[DEBUG] Cluster state has changed!")
 		if err := cluster.Validate(); err != nil {
 			return err
 		}
 		cluster.ModifyRequestOnInstancePool()
 		fixInstancePoolChangeIfAny(d, &cluster)
-		clusterInfo, err = clusters.Edit(cluster)
+
+		numberOfClusterConfigChanges := 0
+		for k := range clusterSchema {
+			if k == "library" || k == "is_pinned" {
+				continue
+			}
+			if d.HasChange(k) {
+				numberOfClusterConfigChanges += 1
+			}
+		}
+		// We can only call the resize api if the cluster is in the running state
+		// and only the cluster size (ie num_workers OR autoscale) is being changed
+		isNumWorkersResizeForNonAutoscalingCluster := numberOfClusterConfigChanges == 1 && d.HasChange("num_workers")
```

Comment:
you can make the code more readable by introducing just two variables: `changedWorkers` & `changedAutoscale`

Reply:
done